### ISSUE TITLE
Question: Can we reload domain and not have error in LateUpdate() in Goap?

### ISSUE THREAD

**Issue by mattemoore** [crashkonijn/GOAP#287]
Reloading domain causes null reference exception 
```
NullReferenceException: Object reference not set to an instance of an object
CrashKonijn.Goap.Runtime.GoapBehaviour.Update () (at ./Library/PackageCache/com.crashkonijn.goap@8639d56f32/Runtime/CrashKonijn.Goap.Runtime/Behaviours/GoapBehaviour.cs:35)
```

![image](https://github.com/user-attachments/assets/d756e3c6-b749-4f52-97ab-9ebc377a151b)

**Package Version**
Beta v3

**Comment by crashkonijn** [comment]
Hi!

Can you please describe exactly what you're doing?

**Comment by mattemoore** [comment]
Sorry I could have been more explicit.

Basically I'm reloading the domain by running the scene in play mode, then changing some code in a Behaviour, saving file, then returning to unity editor which triggers a domain reload (i think?).

**Comment by crashkonijn** [comment]
Should be fixed in 3.0.21-beta!
